pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -585,26 +641,37 @@ func (a *FlavorAssigner) findFlavorForPodSetResource(
 			status.appendf("flavor %s not found", fName)
 			continue
 		}
-		if features.Enabled(features.TopologyAwareScheduling) {
-			if message := checkPodSetAndFlavorMatchForTAS(a.cq, ps, flavor); message != nil {
-				log.Error(nil, *message)
-				status.appendf("%s", *message)
-				continue
+
+		var flavorUnmatchMessage *string
```

Comment:
Instead of using `flavorUnmatchMessage` why don't we just collect the messages in `status` directly, as before?

For one it makes the diff larger, but also feels constraining to have only one message.

Reply:
Thx, for explaining, looking into this and will propose path forward.

Reply:
Hm, so if you just into status, and use break, then you may also get more messages, say for untolerated taints? 

It does not seem like an issue, but I see it would be a change compared to current semantics.

In that case, if we want to preserve the semantics, I would suggest to extract the code which iterates over pod sets `for psIdx, psID := range psIDs {` to a helper function, passing the status there. Once you add the message for one PodSet you can escape downstream logic (and for other PodSets) with return. wdyt?

If you anticipate complications I'm fine with a follow up.